[32mINFO    [0m integration.high_availability.conftest:conftest.py:29 Clearing continuous writes
[32mINFO    [0m integration.high_availability.conftest:conftest.py:32 Starting continuous writes
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:121 Get leader unit
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:126 Run pre-upgrade-check action
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:132 Inject dependency fault
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:137 Refresh the charm
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:140 Wait for upgrade to fail on leader
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:146 Ensure continuous_writes while in failure state
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:342 Ensure continuous writes are incrementing
[32mINFO    [0m integration.helpers:helpers.py:482 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=485 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=485 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=485 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=1432 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=1440 on unit mysql/1
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=1447 on unit mysql/2
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:149 Re-run pre-upgrade-check action
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:152 Re-refresh the charm
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:154 Wait for upgrade to start
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [idle] blocked: ready to rollback application
  mysql/1 [idle] waiting: other units upgrading first...
  mysql/2 [idle] waiting: other units upgrading first...
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] waiting: other units upgrading first...
  mysql/1 [idle] waiting: other units upgrading first...
  mysql/2 [executing] maintenance: starting services...
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [idle] waiting: other units upgrading first...
  mysql/1 [executing] maintenance: recovering unit after upgrade
  mysql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [executing] maintenance: stopping services..
  mysql/1 [idle] maintenance: upgrade completed
  mysql/2 [idle] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:3211 Waiting for model:
  mysql/0 [idle] active: 
  mysql/1 [idle] active: Primary
  mysql/2 [idle] active: 
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:161 Ensure continuous_writes after rollback procedure
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:342 Ensure continuous writes are incrementing
[32mINFO    [0m integration.helpers:helpers.py:482 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=7275 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=7275 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=7298 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=7303 on unit mysql/1
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:367 max_written_value=7308 on unit mysql/2
[32mINFO    [0m integration.high_availability.conftest:conftest.py:37 Clearing continuous writes
[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Model status:

Model    Controller     Cloud/Region         Version  SLA          Timestamp
testing  concierge-lxd  localhost/localhost  3.6.5    unsupported  01:44:11Z

App             Version          Status  Scale  Charm           Channel      Rev  Exposed  Message
mysql           8.0.41-0ubun...  active      3  mysql                          2  no       
mysql-test-app  0.0.2            active      1  mysql-test-app  latest/edge   73  no       

Unit               Workload  Agent      Machine  Public address  Ports           Message
mysql-test-app/0*  active    idle       0        10.252.221.143                  
mysql/0*           active    executing  1        10.252.221.217  3306,33060/tcp  
mysql/1            active    executing  2        10.252.221.96   3306,33060/tcp  Primary
mysql/2            active    executing  3        10.252.221.109  3306,33060/tcp  

Machine  State    Address         Inst id        Base          AZ  Message
0        started  10.252.221.143  juju-53fda7-0  ubuntu@22.04      Running
1        started  10.252.221.217  juju-53fda7-1  ubuntu@22.04      Running
2        started  10.252.221.96   juju-53fda7-2  ubuntu@22.04      Running
3        started  10.252.221.109  juju-53fda7-3  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:868 Juju error logs:

machine-1: 01:31:36 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:31:36 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-0: 01:31:36 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-2: 01:33:33 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:33:33 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-1: 01:33:33 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:33:48 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:33:48 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-2: 01:33:49 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:34:01 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:34:01 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-test-app-0: 01:34:01 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-mysql-0: 01:40:42 ERROR unit.mysql/0.juju-log charms.data_platform_libs.v0.upgrade.VersionError - {'message': 'Versions incompatible, charm 1 can not be upgraded to 2', 'cause': 'Upgrades only supported for specific versions, charm versions satisfying requirement >1', 'resolution': ''}
unit-mysql-0: 01:40:42 CRITICAL unit.mysql/0.juju-log Upgrade failed, follow the instructions below to rollback:
    1. Re-run `pre-upgrade-check` action on the leader unit to enter 'recovery' state
    2. Run `juju refresh` to the previously deployed charm revision or local charm file

[32mINFO    [0m pytest_operator.plugin:plugin.py:947 Forgetting model main...